app/scripts/metamask-controller.js

```diff
@@ -5687,6 +5688,7 @@ export default class MetamaskController extends EventEmitter {
       this.appStateController.store.getState()[appStatePollingTokenType];
     pollingTokensToDisconnect.forEach((pollingToken) => {
       this.gasFeeController.disconnectPoller(pollingToken);
+      this.gasFeeController.stopPollingByPollingToken(pollingToken);
```

Comment:
It looks like this is the new polling API, whereas `disconnectPoller` is the old API. Is that correct? Are we using both of these simultaneously?

Reply:
there are a few references to the old API still

Reply:
@jiexi can you remind me where these references are? These use the same polling tokens?

Reply:
https://github.com/MetaMask/metamask-extension/blob/0a3ef253bbec5c3245078d9bbbf166fbe8b32982/ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js#L43

https://github.com/MetaMask/metamask-extension/blob/0a3ef253bbec5c3245078d9bbbf166fbe8b32982/ui/ducks/send/send.js#L53

Reply:
the poll token from the legacy polling and the networkClientId based polling are not the same

Reply:
correct. calling `disconnectPoller` and `stopPollingByPollingToken` with polling tokens it cannot find is a noop without errors thrown